File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        string res;
        int pn = part.size();

        for (char c : s) {
            res.push_back(c);

            if (res.size() >= pn && res.ends_with(part)) {
                res.erase(res.size() - pn);
            }
        }

        return res;
    }
};

File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        res = []
        pn = len(part)

        for c in s:
            res.append(c)

            while len(res) >= pn and ''.join(res[-pn:]) == part:
                del res[-pn:]

        return "".join(res)